hikari/impl/interaction_server.py

```diff
@@ -0,0 +1,450 @@
+                backlog=backlog,
+                reuse_address=reuse_address,
+                reuse_port=reuse_port,
+            )
+        )
+        loop.run_until_complete(self._future)
+
+    async def start(
+        self,
+        host: typing.Optional[typing.Union[str, typing.Sequence[str]]] = None,
+        port: typing.Optional[int] = None,
+        path: typing.Optional[str] = None,
+        socket: typing.Optional[socket_.socket] = None,
+        shutdown_timeout: float = 60.0,
+        ssl_context: typing.Optional[ssl.SSLContext] = None,
+        backlog: int = 128,
+        reuse_address: typing.Optional[bool] = None,
+        reuse_port: typing.Optional[bool] = None,
+    ) -> None:
+        if self._is_alive:
+            raise TypeError("Cannot start an already running interaction server")
+
+        self._future = asyncio.futures.Future()
+        self._is_alive = True
+        await self._runner.setup()
+        sites: typing.List[aiohttp.web.BaseSite] = []
+
+        if host is not None:
+            if isinstance(host, str):
+                host = [host]
+
+            for h in host:
+                sites.append(
+                    aiohttp.web.TCPSite(
+                        self._runner,
+                        h,
+                        port,
+                        shutdown_timeout=shutdown_timeout,
+                        ssl_context=ssl_context,
+                        backlog=backlog,
+                        reuse_address=reuse_address,
+                        reuse_port=reuse_port,
+                    )
+                )
+
+        elif path is None and socket is None or port is None:
+            sites.append(
+                aiohttp.web.TCPSite(
+                    self._runner,
+                    port=port,
+                    shutdown_timeout=shutdown_timeout,
+                    ssl_context=ssl_context,
+                    backlog=backlog,
+                    reuse_address=reuse_address,
+                    reuse_port=reuse_port,
+                )
+            )
+
+        if path is not None:
+            sites.append(
+                aiohttp.web.UnixSite(
+                    self._runner, path, shutdown_timeout=shutdown_timeout, ssl_context=ssl_context, backlog=backlog
+                )
+            )
+
+        if socket is not None:
+            sites.append(
+                aiohttp.web.SockSite(
+                    self._runner, socket, shutdown_timeout=shutdown_timeout, ssl_context=ssl_context, backlog=backlog
+                )
+            )
+
+        for site in sites:
+            _LOGGER.info("Starting site on %s", site.name)
+            await site.start()
+
+    def set_listener(
+        self, listener: typing.Optional[interaction_server.MainListenerT], /, *, replace: bool = False
+    ) -> None:
+        if not replace and self._listener:
+            raise TypeError("Main listener already set")
+
+        self._listener = listener
```

Comment:
Im not sure this is the correct way to do this, and there are some stuff in here that I don't really like. I'm going to have a think what the best way to implement this is and get back to you

Reply:
correct way to do what specifically